core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2227,7 +2227,7 @@ public AlluxioURI getPath(long fileId) throws FileDoesNotExistException {
 
   @Override
   public Set<Long> getPinIdList() {
-    return mInodeTree.getPinIdSet();
+    return Sets.union(mInodeTree.getPinIdSet(), mInodeTree.getToBePersistedIds());
```

Comment:
shall we unpinn these files after they are persisted?

Reply:
we never pin, so unpinning is not required

Reply:
`InodeTree` handles adding and removing inodes from the set of `ToBePersisted` inodes. This call is used for a snapshot of inodes which should be considered as unevictable.

Reply:
but workers don't know the difference. can worker keep them indefinitely?

Reply:
The worker periodically polls for the current set of pinned inodes, it does not store the state across calls to `getPinIdList`

Reply:
please document this here as the correctness of returning the union of pinned file ids and tobepersisted ids replying on some assumptions externally. Also it looks "confusing" too here to return the union.